src/EFCore.PG/Storage/Internal/Mapping/NpgsqlReadOnlyHstoreTypeMapping.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using Microsoft.EntityFrameworkCore.Storage;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.Mapping
+{
+    /// <summary>
+    /// The type mapping for the PostgreSQL hstore type to immutable .NET Dictionaries.
+    /// </summary>
+    /// <remarks>
+    /// See: https://www.postgresql.org/docs/current/static/hstore.html
+    /// </remarks>
+    public class NpgsqlReadOnlyHstoreTypeMapping : NpgsqlHstoreTypeMapping
```

Comment:
[As discussed](https://github.com/npgsql/npgsql/pull/2784#issuecomment-572535072), while we want to add a mapping for ImmutableDictionary, we shouldn't add mappings to interfaces... So we need to remove this (and everything related elsewhere).